types/src/transaction/mod.rs

```diff
@@ -951,6 +952,56 @@ impl TransactionOutput {
     }
 }
 
+/// Extension of `TransactionOutput` that also holds `DeltaSet`
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct TransactionOutputExt {
+    delta_set: DeltaSet,
+    output: TransactionOutput,
+}
+
+impl TransactionOutputExt {
+    pub fn new(delta_set: DeltaSet, output: TransactionOutput) -> Self {
+        TransactionOutputExt { delta_set, output }
+    }
+
+    pub fn delta_set(&self) -> &DeltaSet {
+        &self.delta_set
+    }
+
+    pub fn write_set(&self) -> &WriteSet {
+        &self.output.write_set
+    }
+
+    pub fn events(&self) -> &[ContractEvent] {
+        &self.output.events
+    }
+
+    pub fn gas_used(&self) -> u64 {
+        self.output.gas_used
+    }
+
+    pub fn status(&self) -> &TransactionStatus {
+        &self.output.status
+    }
+
+    pub fn into(self) -> (DeltaSet, TransactionOutput) {
+        (self.delta_set, self.output)
+    }
+
+    pub fn into_transaction_output(self) -> TransactionOutput {
```

Comment:
how do we expect to use this function? do we expect the delta to be empty here?

Reply:
This function is used when we don't care about deltas, e.g. in `simulate_signed_transaction` but we have `TransactionOutputExt` and need to extract the output. The deltas should be empty here, since no aggregator functionality is exposed there - do you think we need implicit assert/check that we drop empty deltas?

Reply:
we may still have deltas in simulated_signed_transaction right? (like the gas fee) we probably should materialize in the caller in order to return the expected writeset

Reply:
Yes, that's right, I will add a note there